Mage.Sets/src/mage/cards/r/RottenmouthViper.java

```diff
@@ -0,0 +1,204 @@
+                            target.withNotTarget(true);
+                            if (opponent.choose(outcome, target, source, game)) {
+                                Permanent permanent = game.getPermanent(target.getFirstTarget());
+                                if (permanent != null) {
+                                    if (permanent.sacrifice(source, game)) {
+                                        continue;
+                                    }
+                                }
+                            }
+                        }
+                        if (!opponent.getHand().isEmpty() && opponent.chooseUse(outcome, "Discard a card? (Iteration " + i + " of " + repeat + ")",
+                                "Otherwise you lose 4 life.", "Discard", "Lose 3 life", source, game)) {
+                            opponent.discardOne(false, false, source, game);
+                            continue;
+                        }
+                        opponent.loseLife(4, game, source, false);
+                    }
+                }
+            }
+            return true;
+        }
+        return false;
+    }
+}
+
+class RottenmouthViperCostModificationEffect extends CostModificationEffectImpl{
+
+    RottenmouthViperCostModificationEffect(){
+        super(Duration.WhileOnStack, Outcome.Benefit, CostModificationType.REDUCE_COST);
+    }
+
+    private RottenmouthViperCostModificationEffect(final RottenmouthViperCostModificationEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public RottenmouthViperCostModificationEffect copy() {
+        return new RottenmouthViperCostModificationEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source, Ability abilityToModify) {
+        SpellAbility spellAbility = (SpellAbility) abilityToModify;
+        int reduction = 0;
+        for (Cost cost: spellAbility.getCosts()) {
+            if(!(cost instanceof SacrificeXTargetCost)){
+                continue;
+            }
+            if (game.inCheckPlayableState()) {
+                reduction += ((SacrificeXTargetCost) cost).getMaxValue(spellAbility, game);
+            } else{
+                reduction += ((SacrificeXTargetCost) cost).getAmount();
+            }
+            break;
+        }
+        RottenmouthViperCostModificationWatcher watcher = game.getState().getWatcher(RottenmouthViperCostModificationWatcher.class);
+        if (watcher != null) {
+            reduction += watcher.getSacCount(source.getControllerId());
+        }
+        CardUtil.adjustCost(spellAbility, reduction);
+        return true;
+    }
+
+    @Override
+    public boolean applies(Ability abilityToModify, Ability source, Game game) {
+        return abilityToModify instanceof SpellAbility && abilityToModify.getSourceId().equals(source.getSourceId());
+    }
+}
+
+class RottenmouthViperCostModificationWatcher extends Watcher {
```

Comment:
seems overimplemented, Dargo needs a watcher but this doesn't, see Torgaar, Famine Incarnate